Tasks/Common/sonarqube-common/sonarqube-common.ts

```diff
@@ -87,6 +90,19 @@ export function getSonarQubeEndpointFromInput(inputFieldName): SonarQubeEndpoint
     return new SonarQubeEndpoint(hostUrl, hostUsername, hostPassword);
 }
 
+// Returns, as an object, the contents of the 'report-task.txt' file created by SonarQube plugins
+// The returned object contains the following properties:
+//   projectKey, serverUrl, dashboardUrl, ceTaskId, ceTaskUrl
+export function getSonarQubeTaskReport(sonarPluginFolder: string): Map<string, string> {
```

Comment:
We actually use both the dashboard URL and the project title (for the hover text). Using a map is more flexible and is designed specifically to be resistant against file changes. A data class might fulfil many of the same functions (though an implementation of that would probably still go through the Map just for neatness).